torchvision/models/mnasnet.py

```diff
@@ -8,7 +9,7 @@
 
 _MODEL_URLS = {
     "mnasnet0_5":
-    "https://download.pytorch.org/models/mnasnet0.5_top1_67.592-7c6cb539b9.pth",
+    "https://github.com/1e100/mnasnet_trainer/releases/download/0.3/mnasnet0.5_top1_67.823-3ffadce67e.pth",
```

Comment:
Could you elaborate? Older code will still be able to read the older checkpoint, and it's perfectly OK if you keep serving it. Newer code should read the new checkpoint. The checkpoints are not compatible across model versions, so if new code reads an older checkpoint, the model downgrades itself to v1. This is justifiable when you already have a transfer-learned model, but IMO undesirable in case of a pretrained checkpoint.

Reply:
I'm not sure if I agree.

If the user update their version of torchvision, their pre-trained classifier (based on stock MNasNet features from torchvision) will not work anymore, and this is definitely not expected.

Reply:
Not sure I understand. If they load the new weights provided here it will work fine. If they load their old transfer-learned model checkpoint, it will also work fine because the model turns into a v1 model onload. What's the scenario we're trying to guard agains here?

Reply:
If the users only saved the classifier, and not the backbone (which was fixed).
Here is an example:
```python
model = torchvision.models.mnasnet(pretrained=True)
features = model.features
classifier = nn.Linear(2, 2)
# train classifier, not backproping through the feature
# extractor to save compute
with torch.no_grad():
    f = features(input)
output = classifier(f)
loss = F.mse(output, target)
loss.backward()

# and then, save classifier only
torch.save(classifier, 'file.pth')
```
I think this can be a reasonable scenario. And this means that they will need the pre-trained weights for the old models as well.

Reply:
After discussion with @soumith , we decided that we will not take this case into account.
We should just hard-error and replace the old parameters with the new ones.

So I think the current status of the PR is good, thanks!